app/renderer/components/Inspector/HighlighterCentroid.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import InspectorCSS from './Inspector.css';
+import { RENDER_CENTROID_AS } from './shared';
+
+const {CENTROID, OVERLAP, EXPAND} = RENDER_CENTROID_AS;
+
+/**
+ * Shows all element centroids
+ */
+export default class HighlighterCentroid extends Component {
+
+  render () {
+    const {selectedElement = {}, selectHoveredElement, unselectHoveredElement, hoveredElement = {}, selectElement, unselectElement, element, elementProperties,
+           currentRender, hoveredCentroid, selectCentroid, unselectCentroid, selectedCentroid, unselectHoveredCentroid, selectHoveredCentroid} = this.props;
+    const {path: hoveredPath} = hoveredElement;
+    const {path: selectedPath} = selectedElement;
+    const {centerX, centerY, keyCode, path, container} = elementProperties;
+    const centroidClasses = [InspectorCSS['centroid-box']];
+    centroidClasses.push(InspectorCSS[currentRender]);
+
+    // Highlght specific centroids based off of what element user clicks
+    if (currentRender !== EXPAND) {
+      if (hoveredPath === path) {
+        centroidClasses.push(InspectorCSS['hovered-element-box']);
+      }
+      if (selectedPath === path) {
+        centroidClasses.push(InspectorCSS['inspected-element-box']);
+      }
+    }
+
+    if (currentRender !== CENTROID) {
+      if (hoveredCentroid === keyCode) {
+        centroidClasses.push(InspectorCSS['hovered-element-box']);
+      }
+      if (selectedCentroid === keyCode && !element) {
+        centroidClasses.push(InspectorCSS['inspected-element-box']);
+      }
+    }
+
+    const overlapDivStyle = {
+      borderStyle: 'dashed',
+      borderColor: '#178b00',
+      visibility: keyCode === selectedCentroid ? 'visible' : 'hidden'
+    };
+
+    const centroidDivStyle = {
+      left: centerX,
+      top: centerY,
+      borderRadius: element && !container && '0%',
+      ...(currentRender === OVERLAP && overlapDivStyle)
```

Comment:
I'm not quote sure this is going to work as expected if `currentRender === OVERLAP` results to false.
how about `...(currentRender === OVERLAP ? overlapDivStyle : {})`?